src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -19,8 +19,8 @@
          localcacert (get-in config [:puppet-server :localcacert])
          settings    (ca/config->master-settings config)]
 
-     (retrieve-ca-cert! localcacert)
      (initialize-master-ssl! settings certname)
+     (retrieve-ca-cert! localcacert)
```

Comment:
This doesn't make sense to me how this can work now...haven't had a chance to play with the code locally yet though. 

> I needed to ensure that the master initialized its SSL files before (potentially) copying the ca cert.

How come? The partial-state check happens during master initialization, so I would think that swapping these two lines would always result in a partial-state exception since the `:localcacert` won't have been written yet.